### PR TITLE
improve runnableExamples and docs for std/nre

### PR DESCRIPTION
**Pull request by timotheecour**

### CODE REVIEW

File: lib/impure/nre.nim [hunk 1]
```diff
@@ -23,46 +23,38 @@ when defined(js):
...
-    7 .. 7,
-  ]
-  var i = 0
-  for match in "moigagoo".findIter(vowels):
-    doAssert match.matchBounds == expectedResults[i]
-    inc i
+  assert collect(for match in "moiga".findIter(vowels): match.matchBounds) ==
+    @[1 .. 1, 2 .. 2, 4 .. 4]
+  from std/sequtils import toSeq
+  let s = sequtils.toSeq("moiga".findIter(vowels)) # fully qualified to avoid confusion with nre.toSeq
```
Comment (Varriount) [inline lib/impure/nre.nim]: What are the benefits of this change?

File: lib/impure/nre.nim [hunk 2]
```diff
@@ -226,18 +208,6 @@ type
...
-  doAssert "abc".match(re"(\w)").get.captures[0] == "a"
-  doAssert "abc".match(re"(?<letter>\w)").get.captures["letter"] == "a"
-  doAssert "abc".match(re"(\w)\w").get.captures[-1] == "ab"
-
-  doAssert "abc".match(re"(\w)").get.captureBounds[0] == 0 .. 0
-  doAssert 0 in "abc".match(re"(\w)").get.captureBounds == true
-  doAssert "abc".match(re"").get.captureBounds[-1] == 0 .. -1
-  doAssert "abc".match(re"abc").get.captureBounds[-1] == 0 .. 2
-
-
```
Comment (Varriount) [inline lib/impure/nre.nim:240]: Why was this moved?

File: lib/impure/nre.nim (hunk 1 above)
Comment (timotheecour) [reply lib/impure/nre.nim]: collect makes tests shorter and more readable

File: lib/impure/nre.nim (hunk 2 above)
Comment (timotheecour) [reply lib/impure/nre.nim:240]: because match examples belong under match

File: lib/impure/nre.nim (hunk 1 above)
Comment (Varriount) [reply lib/impure/nre.nim]: How so? It certainly makes the code shorter, but now the example requires the reader to know what `collect` does. 

Additionally, placing multiple complex elements all on one line, along with a surrounding function call, is generally considered less readable than spreading them out over multiple lines.

Comment (timotheecour) [reply lib/impure/nre.nim]: `collect` is already used in examples for other modules, eg os, sugar, sequtils, lists, for good reasons. 

> Additionally, placing multiple complex elements all on one line

fixed

**Review by ringabout — APPROVED** [review]
